### PR TITLE
task/02-interfaces Serhii Shevchenko

### PR DESCRIPTION
**Pull request by greyzerua**
- [x] Mentor is added to pr
- [x] AC copied to pr description
- [x] Link to deployed version is added to pr https://greyzerua.github.io/fe-react-2024/
- [x] Interfaces for Product and Category based on [swagger](https://ma-backend-api.mocintra.com/docs#/categories/CategoriesController_getAll) are created and exported in separate file.
- [x] Static header and footer are created according to [mobile](https://www.figma.com/file/mRAOMdJiAsD8fHSVnYwUl9/React-web-page?type=design&node-id=13-17490&mode=design&t=3Thf45kJJYMoxsc0-0) and [desktop](https://www.figma.com/file/MIKTcyr6SvspkZnIQ5nEPD/MA-Courses-2022-2023?type=design&node-id=2471-2253&mode=design&t=4DJ0BIdFZZ4mUgAn-0) designs.
- [x] Shared component SocialIcon is created and used for footer icons
- [x] Link to social network and path to icon are passed to component by the typed props

photo_2024-04-27_13-57-31

### CODE REVIEW

File: src/components/icons/icon.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import type { IconType } from './constants';
+import { Icons } from './constants';
+
+type Props = ComponentProps<'svg'> & {
+    iconType: IconType;
+};
+
+export const Icon = ({ iconType, fill }: Props) => {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
```
Comment (vitalii-mel-work) [inline src/components/icons/icon.tsx]: resolve this error

File: src/components/icons/login-icon.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import type { ComponentProps } from 'react';
+
+export const LoginIcon = ({ stroke = '#FFF' }: ComponentProps<'svg'>) => (
```
Comment (vitalii-mel-work) [inline src/components/icons/login-icon.tsx]: color fff to config or get from theme

File: src/components/social-icon/social-icon.module.css [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  transition: .2s all;
+  display: flex;
+}
+
+.social-icon svg path {
+  transition: .2s all ease-in;
+}
+
+.social-icon:hover svg path {
+  fill: #D43824;
```
Comment (vitalii-mel-work) [inline src/components/social-icon/social-icon.module.css]: recomend create css constants

File: src/interfaces/product.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import type { Category } from './category';
+
+export interface Product {
```
Comment (vitalii-mel-work) [inline src/interfaces/product.ts:3]: IProduct

File: src/components/auth/auth.tsx [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
-        <a href="/login" className={`${styles['auth-buttons__button']} ${styles['auth-buttons__button_login']}`}>
-            <img src={login} alt="" />
+        <Link href="/login" className={`${styles['auth-buttons__button']} ${styles['auth-buttons__button_login']}`}>
+            <Icon iconType={IconType.LOGIN} />
             Login
-        </a>
-        <a href="/sign-up" className={`${styles['auth-buttons__button']} ${styles['auth-buttons__button_sign-up']}`}>
-            <img src={signUp} alt="" />
+        </Link>
+        <Link href="/sign-up" className={`${styles['auth-buttons__button']} ${styles['auth-buttons__button_sign-up']}`}>
```
Comment (vitalii-mel-work) [inline src/components/auth/auth.tsx]: href better get from const , when change url one time, not need replase for all files where use it

File: src/components/icons/constants.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import { BasketIcon } from './basket-icon';
+import { FacebookIcon } from './facebook-icon';
+import { InstagramIcon } from './instagram-icon';
+import { LinkedinIcon } from './linkedin-icon';
+import { LoginIcon } from './login-icon';
+import { MoonIcon } from './moon-icon';
+import { SunIcon } from './sun-icon';
+import { UserIcon } from './user-icon';
+
+export enum IconType {
```
Comment (vitalii-mel-work) [inline src/components/icons/constants.ts]: EIconsType

**Review by vitalii-mel-work — APPROVED** [review]
gj

**Review by vitalii-mel-work — APPROVED** [review]

File: src/interfaces/product.ts (hunk 4 above)
Comment (greyzerua) [reply src/interfaces/product.ts:3]: Link rules does not support such naming of interfaces.

**Review by vitalii-mel-work — APPROVED** [review]
very good job

**Pull request merged**